patterns/1-initial/secure-innersource.md

```diff
@@ -0,0 +1,144 @@
+
+- **PUBLIC** - open source: findable and accessible for all SW developers in the world
+- **INTERNAL** - inner source: findable and accessible for all SW developers in the company
+- **RESTRICTED** - restricted source: findable for all SW developers in the company, low-friction process to gain full access.
+- **CLOSED** - closed source: only accessible to named individuals and not findable via the search engine for other SW developers in the company.
+
+Repositories with RESTRICTED sharing level are included in the [central catalog][innersource-portal] and/or fed into the central search mechanism, together with their `README.md` and some meta information. Their content, however, is not accessible without following a light-weight access request process. This can for example be contacting one of the code owners or some other request mechanism described in the `README.md` file.
+
+Notes:
+
+- This will likely have an impact on contributions due to the additional step to access the code.
+- It is feasible to use a tool that automatically adds the `README.md` and some other meta-data to an internal repository in GitHub, allowing the GitHub search feature to include this data in the search results.
+
+# Resulting Context
+
+*What is the situation after the problem has been solved? The original context is changed indirectly by way of the solution. Often this section can include discussion of the next possible Patterns/problems introduced. This section can be short in content - the solution may not introduce new problems or change much context.*
+
+Many times the InnerSource adoption in an organization will [start as an experiment][start-as-an-experiment], with a small number of SHARED repositories.
+
+If the task force (and with that the Security Team) collaborate in this early phase already, they can make joint experiences with how security-related issues can be identified and addressed. This also allows the Security Teams to build up more confidence on InnerSource from a security angle.
+
+With the increased confidence and lessons learned from this experiment, the task force faces less pushback when rolling out InnerSource to more repositories.
+
+Most importantly, the changes implemented through this pattern lead to more code being shared between teams, and closed source will no longer be the obvious default.
+
+- Definition of sharing levels for repositories, with accompanying guidance of which level to use for what type of repository contents.
+- Increased InnerSource awareness in Security team
+- Increased security awareness in engineering teams
+- Increased collaboration between ISPO, Engineering and Security teams
+- More code will be shared between teams, closed source is no longer the obvious default
+- Concrete implementation of security practices, potentially (partially) automated.
+
+# Known Instances
+
+TBD
```

Comment:
Would this be where you put companies that follow this pattern? If so, my previous employer, Verizon, follows this to a T.

Reply:
@spier

Reply:
@conrogers yes exactly. Theoretically you can just add the company the company name here.

However, it becomes more meaningful if we can add a couple of sentences about how the pattern is used at the given org. For example [as done here](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/2-structured/document-your-guiding-principles.md#known-instances).

Do you still have somebody at Verizon that you would like to pull in here, who can speak to how the pattern is used?

Reply:
Please add Philips. For now without implementation details, these can be added later when certain mechanisms have been implemented and validated.

Reply:
@bartgolsteijn I added Philips.

Reply:
Oh yes, please go ahead and add Verizon. I was there only a few months ago and know the developer and manager over the innersource effort. I'll ask if Kendra would like to add details but truthfully I can add all the detes myself haha.

Reply:
Awesome. Added!